### ISSUE TITLE
My Home Tech Spike: use dynamic content in the mobile app card 

### ISSUE THREAD

**Issue by autumnfjeld** [Automattic/wp-calypso#54441]
## Background
We'd like to better promote the WordPress mobile native apps, as mentioned in the dotcom 3 year plan (p58i-aNO-p2).  
"_Our mobile web experience is on par with our desktop web experience and we’re promoting our native mobile apps in all of the right places_."
My Home can help do this by making better use of the real estate that is currently used to promote the WordPress app. Currently users see this whether they have downloaded the app or not:
![image](https://user-images.githubusercontent.com/5665959/125258205-7463a980-e341-11eb-8cf8-87599947f11f.png)

We can offer more interesting and dynamic content in this area, checking for certain "app states" for example:

1. user never downloaded app (show current content as displayed above)
2. user downloaded, but hasn’t opened app in X days 
3. user downloads and has opened the app - what would we show here?
3. new App features available

The tech spike should determine the feasibility of items

- [x] Can we use tracks data to determine last time user opened app:  `wpios_application_opened` & `wpandroid_application_opened` See references below for examples of how tracks data is being used elesewhere.
- [x] Could we prompt users with general tips about the mobile apps, say 1 week after download, 2 weeks after download (thus we would need conditional logic based on download date) (discuss with @yaelirub )
- [x] How would we know new app features are available, could this be automated? Or would it always require manual content updates?

OUTPUT:
* prototype with card that satisfies  item 2  above

## References
* My Home: Hide 'WordPress App' card on my home if the user already has the mobile app & logged in #52995
* My Home: WordPress App Card on mobile browser #53295
* D62067-code
* Using tracks data: 
  * pbArwn-2nd-p2#comment-3429
  * pbArwn-2nd-p2#comment-3439
  * p1624582765277500-slack-C03DKP1JP

cc: @jeffgolenski

**Comment by autumnfjeld** [comment]
@yaelirub Could you let us know if you have any blockers on your side in terms of getting copy written or any designs done for the items being discussed?   We can start small with this item: 
* user downloaded, but hasn’t opened app in X days

What does your timeline look like? So we know how to prioritize this work.  :)

**Comment by roo2** [comment]
we are recording the last opened time in our database as `mobile_app_last_seen` (by parsing tracks logs), in `parse-logs.php:1375` 👍

**Comment by roo2** [comment]
> Can we use tracks data to determine last time user opened app?

 Yes ✅ we are storing the `mobile_app_last_seen` timestamp by parsing the tracks logs
 
 > Could we prompt users with general tips about the mobile apps, say 1 week after download, 2 weeks after download?

 Yes ✅ we have access to the `issued` field on mobile oAuth tokens which will be the date that the app was installed. Example of accessing the oAuth tokens -> D62067-code
 
> How would we know new app features are available, could this be automated? Or would it always require manual content updates?

I remember an idea was floated about getting them to publish machine reasable release notes, but I think that realistically, we could have some static content that a developer can update when an important feature is realeased, and some logic to show that in the app banner section of my home for _x days_ after it's updated?

**Comment by yaelirub** [comment]
Hey @autumnfjeld , there shouldn't be a blocker but let me get back to you on the timeline

**Comment by yaelirub** [comment]
Ok, @mattmiklic will be able to help with this next week.

**Comment by roo2** [comment]
Ok, so I think what we need to move forward with option two ("_Could we prompt users with general tips about the mobile apps, say 1 week after download, 2 weeks after download_"). We need to clarifyt:
* The exact copy that we want to display
* If their are image assets needed, or is it just text?
* Confirm exactly which time-frames we want to use, e.g.  "_< 1 week_", "_1-2 weeks_", "_> 2 weeks_" were suggested, or we could start with just a single message for all users who have downloaded the app.

 @yaelirub, @mattmiklic can you provide us with these details? also ping me or @autumnfjeld  in the `#dotcom-pod-my-home` a8c slack channel 🙂

**Comment by olaolusoga** [comment]
Let's avoid over-indexing on calling out the mobile app card on my home. Though agree it can be improved, let's make sure there's a balance on the page. cc: @mattmiklic 

Also, think the promo can respond more to context. When a person is on their phone, it makes sense to drive folks to the app. When on desktop is different. We should consider a person's mental model when working on desktop, and nudge them to the app based on moments. My home doesn't need to be all things to all the things.

**Comment by mattmiklic** [comment]
Sharing some quick ideas. These are mostly copy changes to the existing designs, as I agree with Ola's point above that these promos don't need to be overly attention grabbing. That said, we can definitely customize the messaging to fit an individual user's context.

<img width="1865" alt="Screen Shot 2021-07-23 at 1 48 30 PM" src="https://user-images.githubusercontent.com/1396552/126821839-cdb5d756-8629-4efa-81f8-3fd3b56a445e.png">

The **default** shows the existing promo with slightly updated wording. The two **logged out** options are for a user who's downloaded the app but isn't currently logged into it. The **QR code** option is for someone who has the app and is logged in, and just needs a quick link to jump back into it.

We could try all or a subset of these, and I'm happy to iterate on any of these options if there are ideas for how to improve them. I assume we could also try any of this content in the place of the mobile web promo. I didn't mock up what that would look like because it wouldn't really require any new design, it's just a copy change.

Sketch file for specs and design details: https://www.sketch.com/s/6e24ec7d-ed51-4e00-8c44-3a4924f7ebdd

**Comment by jeffgolenski** [comment]
@mattmiklic Awesome. thanks for jumping in with this.

**Comment by roo2** [comment]
@mattmiklic we can't tell if the user has the app or not if they haven't logged in to the app it before. Login sessions in the app never expire so we could only determine if they've logged out manually.

### I think that these things are better to segment users with:

`mobile_app_last_seen`: The last time a logged in user opened the app e.g. _Users who have logged in to the app but haven't opened it for a week_
_Auth Token_: If present, the user is logged in to the app.
_Screen Size_: Is the user on a mobile device, or on desktop. ( We currently use this for the send me an email login link )
_Device Type_: Apple or Android



### Notes from my investigation:

#### We can determine if the user logged into the app but then logged out but that's probably rare
We do get a `mobile_app_last_seen` timestamp which is set when the user last used the app (and was logged in to it)
If we see a user who does have the `mobile_app_last_seen` flag, but who is not logged in, then we can call that a "Logged out" user 👍 
Since (it appears that) the app login session never expires, the user would have had to manually log out to get into this state.

#### We could add code to determine if the user has clicked the promo?
Before the user logs in to the app, we don't know who they are, and when they log out of the app, we're clearing the calypso access token, so it's as if they never had the app installed.
I can see us roughly determining if a user has downloaded the app but hasn't logged in:
* Record when a user clicks the promo
* If the user does not log in to the app, assume that they've downloaded the app but not logged in
That isn't going to be accurate because a lot of users will download the app without clicking the promo, and many users that click the promo won't download the app, but we can't tell.

**Comment by mattmiklic** [comment]
Ok cool. I think we could nix the QR code idea then and just show the versions I labeled "logged out" if we know the user has a token and has downloaded/used the app at some point in the past. We can show the option I labeled "Default" if we have no record of the user having used the app before.

Including the login link if we do know the user has used the app could still be helpful; if they are logged in it will just open the app; if they're not, it'll log them in.

**Comment by roo2** [comment]
ok so we would show the login link if the user has ever used the app, whether they are logged in or not? That makes sense to me if the link will just open the app if they are already logged in. 

Do you know if we have already implemented a similar login link anywhere else, or is this the first time we're doing this?

**Comment by roo2** [comment]
@mattmiklic There were two different options for wording of the mobile login link option:

> Get instant notifications of new comments and likes on your site

or

> Check stats, approve comments, and post whenever inspiration strikes.

Which one do you prefer?

**Comment by yaelirub** [comment]
Hey @roo2  👋    Matt is afk for the week. 
I prefer the latter: "Check stats, approve comments, and post whenever inspiration strikes."
